plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -467,7 +470,24 @@ public static ColumnMapping timestampColumnMapping(TimestampType timestampType)
     public static LongReadFunction timestampReadFunction(TimestampType timestampType)
     {
         checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
-        return (resultSet, columnIndex) -> toPrestoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+        return (resultSet, columnIndex) -> {
+            long epochMicros = toPrestoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+            verifyShortTimestamp(epochMicros, 0, timestampType.getPrecision());
+            return epochMicros;
+        };
+    }
+
+    public static void verifyShortTimestamp(long epochMicros, int picosOfMicro, int precision)
+    {
+        checkArgument(precision <= MAX_SHORT_PRECISION, "Precision must be less then or equal to %s", MAX_SHORT_PRECISION);
+        verify(epochMicros == round(epochMicros, MAX_SHORT_PRECISION - precision), "Invalid value of epochMicros for precision %s: %s", precision, epochMicros);
+        verify(picosOfMicro == 0, "Invalid value of picosOfMicro for precision %s: %s", precision, picosOfMicro);
```

Comment:
This check doesn't make sense. You're passing 0 statically into the function.

however, i'd squash the two verify functions into one:
```
verifyTimestamp(long epochMicros, int picosOfMicro, int precision)
```

Reply:
That was my initial implementation but I got confused by the usage. Since I moved everything into StandardColumNamppings, it's much clearer now. Verification doesn't need to assert precision, since this is done before calling it. Thanks!